conn_test.go

```diff
@@ -1550,6 +1550,36 @@ func TestQuoteIdentifier(t *testing.T) {
 	}
 }
 
+func TestQuoteLiteral(t *testing.T) {
+	var cases = []struct {
+		input string
+		want  string
+	}{
+		{`foo`, `'foo'`},
+		{`foo bar baz`, `'foo bar baz'`},
+		{`foo'bar`, `'foo''bar'`},
+		{`foo\bar`, `E'foo\\bar'`},
+		{`foo\ba'r`, `E'foo\\ba''r'`},
+		{`foo"bar`, `'foo"bar'`},
+		{`foo\x00bar`, `E'foo\\x00bar'`},
+		{`\x00foo`, `E'\\x00foo'`},
+		{`''`, `''''''`},
+		{`'abc'; DROP TABLE users;`, `'''abc''; DROP TABLE users;'`},
+		{`\'`, `E'\\'''`},
+		{`E'\''`, `''''`},
```

Comment:
These cases surprise me. I would prefer that `QuoteLiteral("E''") == "'E'''''"`. That is, a Go string consisting of three ASCII characters is interpreted by PostgreSQL as a literal containing three ASCII characters.